### PR TITLE
handle additional gcc/g++ compiler flags

### LINKED ISSUE #1366: CodeChecker log generated json can't handle filenames with spaces
If I have a file called "lib 2.cpp" it breaks the system as CodeChecker log functionality does not escape the filename in the generated json. The generated command in json is:

```
	"command": "/usr/bin/g++-5 -Wall -c example/lib 2.cpp example/main.cpp"
```
From here, other scripts building on this json break.

### PR DESCRIPTION
**Pull request by gyorb**
resolves #1366

### CODE REVIEW

File: libcodechecker/log/option_parser.py [hunk 1]
```diff
@@ -504,7 +510,7 @@ def skip_n(size):
             return True
 
     # Unhandled compilation argument found.
-    LOG.debug("Unhandled argument: " + str(it.item))
+    LOG.debug("Unhandled gcc argument: %s", str(it.item))
```
Comment (dkrupp) [inline libcodechecker/log/option_parser.py]: Change log message to the meaningful.
Forwarding gcc argument:

**Comment by gyorb** [comment]
Depends on #1508

**Comment by gyorb** [comment]
We will keep the current behavior not to forward unhandled arguments just log them out for debugging.

File: libcodechecker/log/option_parser.py [hunk 2]
```diff
@@ -504,23 +513,20 @@ def skip_n(size):
         if res:
             return True
 
-    # Unhandled compilation argument found.
-    LOG.debug("Unhandled argument: " + str(it.item))
     return False
 
 
-def parse_options(args):
+def parse_options(args, source_files):
```
Comment (Xazax-hun) [inline libcodechecker/log/option_parser.py]: Where do you use the second argument?

File: tests/unit/test_option_parser.py [hunk 3]
```diff
@@ -191,5 +191,42 @@ def test_link_with_include_paths(self):
...
+    def test_unhandled(self):
+        """
+        Unhandled options should be detected as compiler option.
+        """
+        source_files = ["/home/test/main test.cpp"]
+        build_cmd = "g++ -Unhandled " + ' '.join(source_files)
+        res = option_parser.parse_options(build_cmd, source_files)
+        self.assertEqual(res.compile_opts, ["-Unhandled"])
+
+    def test_stuff(self):
```
Comment (martong) [inline tests/unit/test_option_parser.py]: Could you please add more descriptive name for this test? E.g. `test_apostrophe_after_eq_sign`

File: tests/unit/test_option_parser.py [hunk 4]
```diff
@@ -191,5 +191,42 @@ def test_link_with_include_paths(self):
...
+        Unhandled options should be detected as compiler option.
+        """
+        source_files = ["/home/test/main test.cpp"]
+        build_cmd = \
+            'g++ -Unhandled -Dvalue="some value" \"/home/test/main test.cpp\"'
+        res = option_parser.parse_options(build_cmd, source_files)
+        self.assertEqual(res.compile_opts,
+                         ["-Unhandled", "-Dvalue=some value"])
+
+    def test_stuff2(self):
```
Comment (martong) [inline tests/unit/test_option_parser.py]: `test_apostrophe_before_dash` ?

**Comment by gyorb** [comment]
This will be fixed in #1970

**Pull request closed without merging**